src/state/createAnswersState.js

```diff
@@ -63,24 +78,44 @@ class Answers {
         return this.percentageOfCorrect(this.productAnswers);
     }
 
+    @computed get isLoading() {
+        return this.loading.length > 0;
+    }
+
     addAnswer(answer) {
-        this.list.push(answer);
+        this.loading.push(true);
+        api.addAnswer(answer).then(() => {
+            this.list.push(answer);
+            this.loading.pop();
+        })
     }
 
     deleteAnswer(id) {
-        const index = this.list.findIndex(item => item.id === id);
-        index >=0 && this.list.splice(index, 1);
+        this.loading.push(true);
+        api.deleteAnswer(id).then(() => {
+            const index = this.list.findIndex(item => item.id === id);
+            index >=0 && this.list.splice(index, 1);
+            this.unlike(id);
+            this.loading.pop();
```

Comment:
You're doing this correctly, but something to mention that here you cannot modify the state directly since you are not in the action if action strict mode is on.

Reply:
Ok, let's switch to strict mode. But more changes will be required (eg. updating wrong answer)